core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -4343,6 +4335,12 @@ aot_obj_data_create(AOTCompContext *comp_ctx)
         /* close and remove temp file */
         close(fd);
         unlink(file_name);
+#else
+        if (_mktemp_s(file_name, sizeof(file_name))) {
```

Comment:
Same as above, maybe we can add `bh_mkstemp` in bh_common.h/.c? And note there is another similar calling to mkstemp/_mktemp_s in aot_compiler.c aot_generate_tempfile_name, we can call it also.